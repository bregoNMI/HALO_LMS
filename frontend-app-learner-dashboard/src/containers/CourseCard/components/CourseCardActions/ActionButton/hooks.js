import { useWindowSize, breakpoints } from '@openedx/paragon';

export const useIsCollapsed = () => {
  const { width } = useWindowSize();
  return width < breakpoints.medium.maxWidth && width > breakpoints.small.maxWidth;
};

export default useIsCollapsed;
