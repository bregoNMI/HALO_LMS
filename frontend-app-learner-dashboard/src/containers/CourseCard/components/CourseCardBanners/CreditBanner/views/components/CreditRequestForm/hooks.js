import React from 'react';

export const useCreditRequestFormData = (requestData) => {
  const ref = React.useRef(null);
  React.useEffect(() => {
    if (requestData !== null) {
      ref.current.click();
    }
  }, [requestData]);
  return { ref };
};

export default useCreditRequestFormData;
