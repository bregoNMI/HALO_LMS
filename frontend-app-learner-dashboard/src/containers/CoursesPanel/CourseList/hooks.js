import { useWindowSize, breakpoints } from '@openedx/paragon';

export const useIsCollapsed = () => {
  const { width } = useWindowSize();
  return width < breakpoints.medium.maxWidth;
};

export default useIsCollapsed;
