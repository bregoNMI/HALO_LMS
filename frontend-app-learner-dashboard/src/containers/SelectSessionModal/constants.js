/* eslint-disable import/prefer-default-export */
export const LEAVE_OPTION = 'leave';
