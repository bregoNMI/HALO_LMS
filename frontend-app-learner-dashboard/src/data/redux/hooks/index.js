export * from './app';
export * from './requests';
