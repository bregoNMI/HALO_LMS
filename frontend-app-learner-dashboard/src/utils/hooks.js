import { useIntl } from '@edx/frontend-platform/i18n';

import dateFormatter from './dateFormatter';

export const useFormatDate = () => {
  const { formatDate } = useIntl();
  return (date) => dateFormatter(formatDate, date);
};

export default {
  useFormatDate,
};
