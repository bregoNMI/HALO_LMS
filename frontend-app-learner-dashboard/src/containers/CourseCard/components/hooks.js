import { reduxHooks } from 'hooks';

export const useActionDisabledState = (cardId) => {
  const { isMasquerading } = reduxHooks.useMasqueradeData();
  const {
    canUpgrade, hasAccess, isAudit, isAuditAccessExpired,
  } = reduxHooks.useCardEnrollmentData(cardId);
  const {
    isEntitlement, isFulfilled, canChange, hasSessions,
  } = reduxHooks.useCardEntitlementData(cardId);

  const { resumeUrl, homeUrl, upgradeUrl } = reduxHooks.useCardCourseRunData(cardId);

  const disableBeginCourse = !homeUrl || (isMasquerading || !hasAccess || (isAudit && isAuditAccessExpired));
  const disableResumeCourse = !resumeUrl || (isMasquerading || !hasAccess || (isAudit && isAuditAccessExpired));
  const disableViewCourse = !hasAccess || (isAudit && isAuditAccessExpired);
  const disableSelectSession = !isEntitlement || isMasquerading || !hasAccess || (!canChange || !hasSessions);
  const disableUpgradeCourse = !upgradeUrl || (isMasquerading && !canUpgrade);

  const disableCourseTitle = (isEntitlement && !isFulfilled) || disableViewCourse;

  return {
    disableBeginCourse,
    disableResumeCourse,
    disableViewCourse,
    disableUpgradeCourse,
    disableSelectSession,
    disableCourseTitle,
  };
};

export default useActionDisabledState;
