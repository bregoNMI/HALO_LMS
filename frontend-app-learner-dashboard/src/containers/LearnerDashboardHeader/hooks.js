import React from 'react';
import { useWindowSize, breakpoints } from '@openedx/paragon';
import track from 'tracking';
import { StrictDict } from 'utils';
import { linkNames } from 'tracking/constants';

import * as module from './hooks';

export const state = StrictDict({
  isOpen: (val) => React.useState(val), // eslint-disable-line
});

export const useIsCollapsed = () => {
  const { width } = useWindowSize();
  const isCollapsed = React.useMemo(() => (width <= breakpoints.large.minWidth), [width]);
  return isCollapsed;
};

export const findCoursesNavClicked = (href) => track.findCourses.findCoursesClicked(href, {
  linkName: linkNames.learnerHomeNavExplore,
});

export const findCoursesNavDropdownClicked = (href) => track.findCourses.findCoursesClicked(href, {
  linkName: linkNames.learnerHomeNavDropdownExplore,
});

export const useLearnerDashboardHeaderData = () => {
  const [isOpen, setIsOpen] = module.state.isOpen(false);
  const toggleIsOpen = () => setIsOpen(!isOpen);

  return {
    isOpen,
    toggleIsOpen,
  };
};

export default {
  useIsCollapsed,
  findCoursesNavClicked,
  findCoursesNavDropdownClicked,
  useLearnerDashboardHeaderData,
};
