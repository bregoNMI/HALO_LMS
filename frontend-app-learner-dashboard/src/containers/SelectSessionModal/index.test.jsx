import React from 'react';
import { shallow } from '@edx/react-unit-test-utils';

import hooks from './hooks';
import SelectSessionModal from '.';

jest.mock('./hooks', () => ({
  __esModule: true,
  default: jest.fn(),
}));

const hookReturn = {
  availableSessions: [],
  showModal: true,
  closeSessionModal: jest.fn().mockName('useSelectSessionModalData.closeSessionModal'),
  showLeaveOption: true,
  header: 'test-header',
  hint: 'test-hint',
};

const availableSessions = [
  { startDate: '1/2/2000', endDate: '1/2/2020', courseId: 'test-course-id-1' },
  { startDate: '2/3/2000', endDate: '2/3/2020', courseId: 'test-course-id-2' },
  { startDate: '3/4/2000', endDate: '3/4/2020', courseId: 'test-course-id-3' },
];

describe('SelectSessionModal', () => {
  describe('snapshot', () => {
    test('empty modal with leave option ', () => {
      hooks.mockReturnValueOnce({
        ...hookReturn,
      });
      expect(shallow(<SelectSessionModal />).snapshot).toMatchSnapshot();
    });

    test('modal with leave option ', () => {
      hooks.mockReturnValueOnce({
        ...hookReturn,
        availableSessions: [...availableSessions],
      });
      expect(shallow(<SelectSessionModal />).snapshot).toMatchSnapshot();
    });

    test('modal without leave option ', () => {
      hooks.mockReturnValueOnce({
        ...hookReturn,
        availableSessions,
        showLeaveOption: false,
      });
      expect(shallow(<SelectSessionModal />).snapshot).toMatchSnapshot();
    });
  });
});
