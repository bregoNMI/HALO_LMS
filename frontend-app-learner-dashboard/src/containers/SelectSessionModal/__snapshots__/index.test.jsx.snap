// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectSessionModal snapshot empty modal with leave option  1`] = `
<ModalDialog
  className="p-4 px-4.5"
  hasCloseButton={false}
  isFullscreenOnMobile={true}
  isOpen={true}
  onClose={[MockFunction utils.nullMethod]}
  size="md"
  title="test-header"
>
  <h3>
    test-header
  </h3>
  <Form.Group
    className="pt-3"
  >
    <Form.Label>
      test-hint
    </Form.Label>
    <Form.RadioSet
      className="pt-3 pb-4"
      name="sessions"
    >
      <Form.Radio
        value="leave"
      >
        Leave session
      </Form.Radio>
    </Form.RadioSet>
  </Form.Group>
  <ActionRow>
    <Button
      onClick={[MockFunction useSelectSessionModalData.closeSessionModal]}
      variant="tertiary"
    >
      Never mind
    </Button>
    <Button>
      Confirm Session
    </Button>
  </ActionRow>
</ModalDialog>
`;

exports[`SelectSessionModal snapshot modal with leave option  1`] = `
<ModalDialog
  className="p-4 px-4.5"
  hasCloseButton={false}
  isFullscreenOnMobile={true}
  isOpen={true}
  onClose={[MockFunction utils.nullMethod]}
  size="md"
  title="test-header"
>
  <h3>
    test-header
  </h3>
  <Form.Group
    className="pt-3"
  >
    <Form.Label>
      test-hint
    </Form.Label>
    <Form.RadioSet
      className="pt-3 pb-4"
      name="sessions"
    >
      <Form.Radio
        key="test-course-id-1"
        value="test-course-id-1"
      >
        1/2/2000
         - 
        1/2/2020
      </Form.Radio>
      <Form.Radio
        key="test-course-id-2"
        value="test-course-id-2"
      >
        2/3/2000
         - 
        2/3/2020
      </Form.Radio>
      <Form.Radio
        key="test-course-id-3"
        value="test-course-id-3"
      >
        3/4/2000
         - 
        3/4/2020
      </Form.Radio>
      <Form.Radio
        value="leave"
      >
        Leave session
      </Form.Radio>
    </Form.RadioSet>
  </Form.Group>
  <ActionRow>
    <Button
      onClick={[MockFunction useSelectSessionModalData.closeSessionModal]}
      variant="tertiary"
    >
      Never mind
    </Button>
    <Button>
      Confirm Session
    </Button>
  </ActionRow>
</ModalDialog>
`;

exports[`SelectSessionModal snapshot modal without leave option  1`] = `
<ModalDialog
  className="p-4 px-4.5"
  hasCloseButton={false}
  isFullscreenOnMobile={true}
  isOpen={true}
  onClose={[MockFunction utils.nullMethod]}
  size="md"
  title="test-header"
>
  <h3>
    test-header
  </h3>
  <Form.Group
    className="pt-3"
  >
    <Form.Label>
      test-hint
    </Form.Label>
    <Form.RadioSet
      className="pt-3 pb-4"
      name="sessions"
    >
      <Form.Radio
        key="test-course-id-1"
        value="test-course-id-1"
      >
        1/2/2000
         - 
        1/2/2020
      </Form.Radio>
      <Form.Radio
        key="test-course-id-2"
        value="test-course-id-2"
      >
        2/3/2000
         - 
        2/3/2020
      </Form.Radio>
      <Form.Radio
        key="test-course-id-3"
        value="test-course-id-3"
      >
        3/4/2000
         - 
        3/4/2020
      </Form.Radio>
    </Form.RadioSet>
  </Form.Group>
  <ActionRow>
    <Button
      onClick={[MockFunction useSelectSessionModalData.closeSessionModal]}
      variant="tertiary"
    >
      Never mind
    </Button>
    <Button>
      Confirm Session
    </Button>
  </ActionRow>
</ModalDialog>
`;
